### ISSUE TITLE
Visdom Overlapping Images.

### ISSUE THREAD

**Issue by mightyabra** [facebookresearch/visdom#664]
I am working on a neural network code and i am using Visdom server for my visualization. Everything was working well until last two days, my confusion matrix started replicating itself and covering my other graphs (test graph, test loss graph). My training accuracy graph is now totally messed up. I now have three confusion matrix instead of one. 

Software used: Anaconda (using Pytorch)
OS: Windows 10
Browser: Chrome

1. I have uninstalled and reinstalled Anaconda 

2. I uninstalled and reinstalled visdom.

3. i seem not to know what i am doing wrong.

4. I am using the code on a different machine and it is working well (as in all the graphs are showing correctly and in their place).

![Visdom Error](https://user-images.githubusercontent.com/40210809/64063860-af471300-cbe9-11e9-891a-87807830aca0.PNG)

![code](https://user-images.githubusercontent.com/40210809/64063869-fc2ae980-cbe9-11e9-8bf9-c6c5b9d3bab7.PNG)

**Comment by JackUrb** [comment]
Hi @mightyabra - unfortunately I won't be able to debug what's going on here without knowing what `VisdomPlotLogger` and `VisdomLogger` are, as the bug is likely there. The behavior you're describing would be what happens if the logger started confusing the unique window ids being sent to the visdom package.

**Comment by mightyabra** [comment]
VisdomPlotLogger and VisdomLogger are torchnet modules for logging outputs to Visdom. Please take a look at this website https://buildmedia.readthedocs.org/media/pdf/tnt/latest/tnt.pdf.
Its a documentation about VisdomLogger and PlotLogger functions.

**Comment by JackUrb** [comment]
Sorry, I don't really know anything about this implementation, but from what I could see in a first pass it is doing something potentially strange to keep track of window ids. You may have more luck using it by specifying a `win=<>` argument to each logger that you specify a unique id such as `'train_loss_win'`.

**Comment by mightyabra** [comment]
thanks @JackUrb  let me try that out and give you feedback.

**Comment by davidebaltieri31** [comment]
Had the same thing happen to me. When a window id is not provided visdom creates one using, among other things, a function called get_rand_id. I think that function returns the same string if it is called twice in less than 1 millisecond. get_rand_id uses time.time() which might have a very, system dependent, bad resolution. Always specifying the 'win' params solves the issue.

**Comment by JackUrb** [comment]
@davidebaltieri31 - you're right, I had been meaning to update the method of generating window ids for a while, but just never got around to it. Happy to accept a PR with the fix.

**Comment by mightyabra** [comment]
Thanks guys for your suggestions.
